src/ai/backend/manager/idle.py

```diff
@@ -368,6 +371,10 @@ async def populate_config(self, raw_config: Mapping[str, Any]) -> None:
         }
         self.thresholds_check_operator = config.get("thresholds-check-operator")
         self.time_window = config.get("time-window")
+        self.initial_grace_period = config.get("initial-grace-period")
+        self.time_pass_grace_period = self.initial_grace_period.total_seconds() \
+            + (await self._redis.time())
+        self.grace_period_flag = False
```

Comment:
Hmm... The grace period should be applied session-wise. Each session should not be terminated at least for the duration of the grace period. Since `populate_config` is called only once at the start of the Manager, every session is subject to be garbage collected after the `initial-grace-period` passed upon Manager restart. To get the per session grace timeout, you may save the information on Redis or just reference the session's `created_at` information.

Also, `self.grace_period_flag` has no role in the current code.